Kernel/Devices/GPU/DisplayConnector.cpp

```diff
@@ -80,7 +80,7 @@ ErrorOr<void> DisplayConnector::allocate_framebuffer_resources(size_t rounded_si
     if (!m_framebuffer_at_arbitrary_physical_range) {
         VERIFY(m_framebuffer_address.value().page_base() == m_framebuffer_address.value());
         m_shared_framebuffer_vmobject = TRY(Memory::SharedFramebufferVMObject::try_create_for_physical_range(m_framebuffer_address.value(), rounded_size));
```

Comment:
`SharedFramebufferVMObject::try_create_for_physical_range` will currently create an `AnonymousVMObject`. 
It should probably also use a `MMIOVMObject` now if this constructor is used

Reply:
It's not obvious to me if it should, it does a bit too much trickery with switching between regions/vmobjects etc for me to be sure. @supercomputer7 what do you think?

Reply:
The Raspberry Pi framebuffer is in the VideoCore memory, which is after RAM.
So if we want to use the devicetree for the memory map on aarch64, we would have to make the SharedFramebufferVMObject work with memory after RAM.

Reply:
Generally, I would agree, but there's a bunch of logic in SharedFramebufferVMObject that deals with mapping and remapping the backing regions live (using the PhysicalPages), which I don't think would currently work with MMIOVMObject.
TBH, I'm not sure all of this logic has to be this complicated, but that's an issue for another time.

Reply:
If we want to support virtual consoles and graphical desktop, we have to somehow ensure that when you move from a graphical TTY to a non-graphical one (or the other way around), then either the VT code gets access or the WindowManager is able to print pixels. This is why it's called a shared buffer, but only one client has access to it each time.

as for the actual implementation, it could be MMIO-oriented, although some GPUs don't do this and don't really allow MMIO access to the framebuffer, but rather you need to initialize a DMA transfer somehow.
But for most of our implementations we should back the shared buffer VMObject with an anonymous vmobject for shadow copy, and MMIO one for actual access.